### PR TITLE
Add generic rolling window function

### PR DESCRIPTION
**Pull request by dwillmer**
- `windowed_pass` - API / function name is subject to change, all recommendations welcome.

### CODE REVIEW

**Comment by dwillmer** [comment]
Should be merged after #20 and #21, built on the same commits

**Comment by dwillmer** [comment]
Added a rolling ols example + unit-tests.

Needs a wrapping function to work due to the multiple args, but that actually works out as an advantage when it comes to t-stats/r^2.

Unfortunately this raised an issue with raw numpy functions being used in the ols functions - we need to call them as module-level functions to ensure they don't get re-parsed. Ufuncs / c-api / cython functions will just error when re-parsed

**Comment by dwillmer** [comment]
Added an example to show multiple return values from a function - t-stats and r^2 in a rolling OLS.

**Comment by pawroman** [comment]
Current PR makes it hard to review due to changes pulled in from other branch

**Comment by dwillmer** [comment]
Yea, agreed, let's merge the others first.

**Comment by pawroman** [comment]
Need to resolve conflicts and should be good to review.

**Comment by dwillmer** [comment]
@fastats/core  - conflicts resolved, please review.

File: fastats/core/windowed_pass.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    for i in range(win, x.shape[0]+1):
+        res = value(x[i-win:i])
+        for j, j_val in enumerate(res):
+            result[i-1, j] = j_val
+    return result
+
+
+if __name__ == '__main__':
+    import pytest
+    pytest.main()
```
Comment (pawroman) [inline fastats/core/windowed_pass.py]: `pytest.main([__file__])`

**Review by pawroman — CHANGES_REQUESTED** [review]
Looks good, just one `pytest.main()` call to fix. Also, I'm wondering how it's passing the coverage checks.

File: tests/core/test_windowed_pass.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+    assert np.isnan(res[0])
+    assert np.isnan(res[1])
+    assert np.isnan(res[2])
+    assert np.isnan(res[3])
+    assert res[4] == approx(5.89915248)
+    assert res[5] == approx(8.64869932)
+    assert res[-1] == approx(274.36253389)
+
+    raw = windowed_pass(data, 5)
```
Comment (pawroman) [inline tests/core/test_windowed_pass.py:28]: Is this how you make coverage work?

Comment (dwillmer) [reply tests/core/test_windowed_pass.py:28]: yea, call it without substitutions

File: fastats/core/windowed_pass.py (hunk 1 above)
Comment (dwillmer) [reply fastats/core/windowed_pass.py]: Thanks, fixed!

**Review by pawroman — APPROVED** [review]

**Pull request merged**